tests/rptest/tests/license_enforcement_test.py

```diff
@@ -201,3 +201,27 @@ def test_enterprise_cluster_bootstrap(self, root_driven_bootstrap):
                                  timeout_sec=60,
                                  backoff_sec=1,
                                  err_msg="The cluster hasn't stabilized")
+
+    @cluster(num_nodes=5)
+    @matrix(enable_cloud_storage=[False, True])
+    def test_cloud_storage_topic_params(self, enable_cloud_storage):
+        if enable_cloud_storage:
+            si_settings = SISettings(self.test_context)
+            self.redpanda.set_si_settings(si_settings)
+
+        super().setUp()
+
+        self.redpanda.set_environment(
+            {'__REDPANDA_DISABLE_BUILTIN_TRIAL_LICENSE': True})
+        self.redpanda.restart_nodes(self.redpanda.nodes)
+        self.redpanda.wait_until(self.redpanda.healthy,
+                                 timeout_sec=60,
+                                 backoff_sec=1,
+                                 err_msg="The cluster hasn't stabilized")
+
+        try:
+            self.rpk.create_topic("test",
+                                  config={"redpanda.remote.write": "true"})
+            assert not enable_cloud_storage, "Should have failed to create topic with redpanda.remote.write set and cloud_storage_enabled set to True"
```

Comment:
is this the entirety of the issue? i vaguely understood that there was a question of whether upgrade was possible if 
- cloud_storage_enabled = false
- topics _already existing_ w/ remote_read or whatever set to true

Maybe I misunderstood though

Reply:
The upgrade should be possible - I can add in a test for that if you like